static/src/javascripts/projects/commercial/modules/cmp/cmp.js

```diff
@@ -0,0 +1,216 @@
+        } else if (
+            !this.store.vendorConsentData &&
+            (command === 'getVendorConsents' || command === 'getConsentData')
+        ) {
+            // Special case where we have the full CMP implementation loaded but
+            // we still queue these commands until there is data available.
+            log.info(
+                `Queuing command: ${command} until consent data is available`
+            );
+            this.commandQueue.push({
+                command,
+                parameter,
+                callback,
+            });
+        } else {
+            log.info(`Proccess command: ${command}, parameter: ${parameter}`);
+            this.commands[command](parameter, callback);
+        }
+    };
+
+    processCommandQueue = () => {
+        const queue = [...this.commandQueue];
+        if (queue.length) {
+            log.info(`Process ${queue.length} queued commands`);
+            this.commandQueue = [];
+            queue.forEach(({ callId, command, parameter, callback, event }) => {
+                if (event) {
+                    this.processCommand(command, parameter, returnValue =>
+                        event.source.postMessage(
+                            {
+                                __cmpReturn: {
+                                    callId,
+                                    command,
+                                    returnValue,
+                                },
+                            },
+                            event.origin
+                        )
+                    );
+                } else {
+                    this.processCommand(command, parameter, callback);
+                }
+            });
+        }
+    };
+    // $FlowFixMe
+    receiveMessage = ({ data, origin, source }) => {
+        const { __cmpCall: cmp } = data;
+        if (cmp) {
+            const { callId, command, parameter } = cmp;
+            this.processCommand(command, parameter, returnValue =>
+                source.postMessage(
+                    { __cmpReturn: { callId, command, returnValue } },
+                    origin
+                )
+            );
+        }
+    };
+
+    notify = (event: string, data: any): void => {
+        log.info(`Notify event: ${event}`);
+        const eventSet = this.eventListeners[event] || [];
+        eventSet.forEach(listener => {
+            listener({ event, data });
+        });
+        // Process any queued commands that were waiting for consent data
+        if (event === 'onSubmit') {
+            this.processCommandQueue();
+        }
+    };
+}
+
+export const init = (): void => {
+    // Only run our CmpService if prepareCmp has added the stub
+    if (window[CMP_GLOBAL_NAME]) {
+        // Pull queued command from __cmp stub
+        const { commandQueue = [] } = window[CMP_GLOBAL_NAME] || {};
+        // Initialize the store with all of our consent data
+        const store = new CmpStore(
```

Comment:
Is that the link between CMP and our own consent?

Reply:
I have added a stub window.__cmp in prepareCmp.scala.js... which is loaded as part of the nonblocking JS so it can get on the page ASAP (suggested by the framework docs). The stub is also responsible for setting the event listeners so iframe messages can be sent. The CMPService in commercial modules then completes the functionality, and now that the API is all there it will run through any requests that may have queued up in the interim.

Store is where I am keeping most of the state, so we only have to read from the cookie and construct the vendorConsentData once. Does that answer the question?

Reply:
Yep. I suppose what I want to understand is how this prebid/AIB consent keeps in sync with our own cookie value. Is this Store built each time freshly so the cookie is always the source of truth?